### DISCUSSION TITLE
Error Found

### DISCUSSION THREAD

**Discussion by PhD-GOAT** [raphaelvallat/yasa#66] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/93689515/168504488-89172b49-da32-4064-9eb0-52c1be144988.png)

```import pandas as pd
import mne
import yasa

raw_file = 'C:/yasa_example_night_young.edf'
hypno_file = 'C:/yasa_example_night_young_hypno.csv'

raw = mne.io.read_raw_edf(raw_file, preload=True)
hypno= pd.read_csv(hypno_file)

hypno_up = yasa.hypno_upsample_to_data(hypno, sf_hypno=1/30, data=raw)
print(len(hypno_up))

yasa.bandpower(raw)
bandpower = yasa.bandpower(raw, hypno=hypno_up, include=(2, 3, 4))
yasa.bandpower(raw, relative=False, bands=[(1, 9, "Slow"), (9, 30, "Fast")])
bandpower.to_csv("bandpower.csv")
fig = yasa.topoplot(bandpower.xs(3)['Delta'])```



It didn't go well while doing it with data other than the sample data of Yasa. So I tried it with the sample data of Yasa, and an error appears.

**Comment by remrama** [comment, marked as answer]
Try putting `squeeze=True` as an argument inside `pd.read_csv`.

Thanks for including your code. You can see the issue is where yasa checks to make sure you are using a list, numpy array, or pandas series for your hypnogram. I suspect you have a 1-column dataframe instead. There are many (easy) solutions if that's the problem, inserting the squeeze argument is just one.

**Reply by PhD-GOAT** [reply]
![image](https://user-images.githubusercontent.com/93689515/168505494-06ae1ea5-024d-49a8-b19d-d8b6381f132e.png)

https://raphaelvallat.com/yasa/build/html/quickstart.html#id11

I just copied and pasted it here, but it doesn't work

**Reply by remrama** [reply]
Ok, but it _did_ work.

This is a new error. So putting squeeze in solved your problem and now you have a new problem. Is that correct?

**Reply by PhD-GOAT** [reply]
I put in the squeeze.
And if I use the data I want to use instead of the sample data, the data types of hypno and hypno_up are different. It's different in the sample of Yasa, so what's going on?

**Reply by remrama** [reply]
Slow down.

Have you solved the problem that you originally posted about? You posted that you found an error. Your code wouldn't run because it errored out when you ran `yasa.hypno_upsample_to_data`. I proposed a solution and you tried it. Now you have responded with a *new problem*, so should I assume your original question was answered? I will.

So you have a **new** error. With this one, yasa is nicely telling you what's going on.

> what's going on?

`The following electrodes have overlapping positions...`
Have you tried addressing this error message from yasa at the bottom of your screenshot?

**Reply by PhD-GOAT** [reply]
That's right. I'm faced with a new problem. But I didn't understand what was going on because I thought it was working well on the Yasa sample, but it didn't work as it was.

**Reply by remrama** [reply]
Ok great.

> Have you tried addressing this error message from yasa at the bottom of your screenshot?

**Reply by PhD-GOAT** [reply]
I don't know how to solve that message now because I did it with the example code. I'm going to look for it after I eat.

**Reply by PhD-GOAT** [reply]
`raw.drop_channels(['ROC-A1', 'LOC-A2', 'EMG1-EMG2', 'EKG-R-EKG-L'])`

It's the same thing! But to get to the point, why does the dtype problem occur? Even if hypno and hypno_up are different, they work in the sample.
`bandpower = yasa.bandpower(raw, hypno=hypno_up, include=(2, 3, 4))`
`AssertionError: hypno and include must have same dtype`

**Reply by raphaelvallat** [reply]
@PhD-GOAT what is the dtype of `hypno_up`? you can check with `hypno_up.dtype`

**Reply by PhD-GOAT** [reply]
![image](https://user-images.githubusercontent.com/93689515/168929594-313a157c-63c1-4d68-8c65-9b3cd2937f10.png)

There was no such error in your sample. And when I do dtype, it says object exactly the same.
